x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/LongScriptFieldScript.java

```diff
@@ -32,14 +32,47 @@
     }
 
     public interface LeafFactory {
-        LongScriptFieldScript newInstance(LeafReaderContext ctx, LongConsumer sync) throws IOException;
+        LongScriptFieldScript newInstance(LeafReaderContext ctx) throws IOException;
     }
 
-    private final LongConsumer sync;
+    private long[] values = new long[1];
+    private int count;
 
-    public LongScriptFieldScript(Map<String, Object> params, SearchLookup searchLookup, LeafReaderContext ctx, LongConsumer sync) {
+    public LongScriptFieldScript(Map<String, Object> params, SearchLookup searchLookup, LeafReaderContext ctx) {
         super(params, searchLookup, ctx);
-        this.sync = sync;
+    }
+
+    /**
+     * Execute the script for the provided {@code docId}.
+     */
+    public final void runForDoc(int docId) {
+        count = 0;
+        setDocument(docId);
+        execute();
+    }
+
+    /**
+     * Values from the last time {@link #runForDoc(int)} was called. This array
+     * is mutable and will change with the next call of {@link #runForDoc(int)}.
+     * It is also oversized and will contain garbage at all indices at and
+     * above {@link #count()}.
+     */
+    public final long[] values() {
+        return values;
+    }
+
+    /**
+     * The number of results produced the last time {@link #runForDoc(int)} was called.
+     */
+    public final int count() {
+        return count;
+    }
+
+    protected final void add(long v) {
```

Comment:
does it need to be protected? Also maybe rename to something like collectValue ? I find it weird to call add against the script itself

Reply:
> does it need to be protected?

I thought so, but I'll take a look.



> Also maybe rename to something like collectValue

:+1: